.azure-pipelines.yml

```diff
@@ -0,0 +1,248 @@
+    - bash: RUST_BACKTRACE=1 cargo test --release --package lightbeam
+      displayName: Cargo test lightbeam release
+      # Lightbeam tests fail right now, but we don't want to block on that.
+      continueOnError: true
+      condition: and(eq(variables['rustup_toolchain'], 'nightly'), eq( variables['build_type'], 'release' ))
+
+    - script: sccache --show-stats
+
+    # Only run on macOS builders, as those are fastest
+    - bash: |
+        cargo doc
+      displayName: Test docs generation
+      condition: and(eq( variables['Agent.OS'], 'Darwin' ), eq( variables['build_type'], 'release' ))
+
+    - bash: |
+        mkdir $BUILD_BINARIESDIRECTORY/bundle
+        cp target/release/{wasmtime,wasm2obj}.exe $BUILD_BINARIESDIRECTORY/bundle
+      displayName: Copy binaries (Win)
+      condition: and(eq( variables['Agent.OS'], 'Windows_NT' ), eq( variables['build_type'], 'release' ))
+    - bash: |
+        mkdir $BUILD_BINARIESDIRECTORY/bundle
+        cp target/release/{wasmtime,wasm2obj} $BUILD_BINARIESDIRECTORY/bundle
+      displayName: Copy binaries (*nix)
+      condition: and(ne( variables['Agent.OS'], 'Windows_NT' ), eq( variables['build_type'], 'release' ))
+    - task: ArchiveFiles@2
+      inputs:
+        rootFolderOrFile: $(Build.BinariesDirectory)/bundle
+        includeRootFolder: false
+        archiveType: 'zip'
+        archiveFile: '$(Build.ArtifactStagingDirectory)/wasmtime-windows-64bit-$(tagName).zip'
+      displayName: Archive files (Win)
+      condition: and(eq( variables['Agent.OS'], 'Windows_NT' ), eq( variables['build_type'], 'release' ))
+    - task: ArchiveFiles@2
+      inputs:
+        rootFolderOrFile: $(Build.BinariesDirectory)/bundle
+        includeRootFolder: false
+        archiveType: 'tar'
+        tarCompression: 'xz'
+        archiveFile: '$(Build.ArtifactStagingDirectory)/wasmtime-macos-64bit-$(tagName).tar.xz'
+      displayName: Archive files (macOS)
+      condition: and(eq( variables['Agent.OS'], 'Darwin' ), eq( variables['build_type'], 'release' ))
+    - task: ArchiveFiles@2
+      inputs:
+        rootFolderOrFile: $(Build.BinariesDirectory)/bundle
+        includeRootFolder: false
+        archiveType: 'tar'
+        tarCompression: 'xz'
+        archiveFile: '$(Build.ArtifactStagingDirectory)/wasmtime-linux-64bit-$(tagName).tar.xz'
+      displayName: Archive files (Linux)
+      condition: and(eq( variables['Agent.OS'], 'Linux' ), eq( variables['build_type'], 'release' ))
+    - task: PublishPipelineArtifact@1
+      inputs:
+        path: $(Build.ArtifactStagingDirectory)/
+        artifactName: 'bundle-$(Agent.OS)'
+      condition: eq( variables['build_type'], 'release' )
+
+- job: Publish
+  dependsOn: Build
+  condition: and(succeeded(), in(variables['Build.Reason'], 'IndividualCI', 'BatchedCI'))
+  steps:
+  - checkout: none
+  - task: DownloadPipelineArtifact@1
+    inputs:
+      targetPath: $(Build.ArtifactStagingDirectory)
```

Comment:
I'm actually sort of surpsied that this works, I thought you had to list each pipeline artifact. I guess this downloads all of them by default?

Reply:
Exactly!